### PR TITLE
feat: add probe and scrapeConfig selectors

### PR DESCRIPTION
**Pull request by carlkyrillos**
feat: ObO sets the `probeSelector` and `probeNamespaceSelector` fields in the Prometheus `.spec`
feat: ObO sets the `scrapeConfigSelector` and `scrapeConfigNamespaceSelector` fields in the Prometheus `.spec`

fixes: #283

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Developer Docs
 
-In order to contribute to this project, make sure you have go 1.18 installed
+In order to contribute to this project, make sure you have go 1.20 installed
```
Comment (sthaha) [inline docs/developer.md:3]: Just a note for next time ... 
Please ensure that a commit does not add an unrelated change.

**Review by simonpasquier — APPROVED** [review]
LGTM, thanks!
